src/coreclr/src/tools/Common/TypeSystem/Common/TypeSystemException.cs

```diff
@@ -13,6 +16,9 @@ public abstract class TypeSystemException : Exception
     {
         private string[] _arguments;
 
+        private static Lazy<ResourceManager> s_stringResourceManager =
```

Comment:
Would it be possible to pull this out of `TypeSystemException` into a separate file? This will not work the the type system is used as a runtime type system (we need to be able to get rid of `ResourceManager` in the [UseSystemResourceKeys](https://github.com/dotnet/runtime/blob/master/docs/workflow/trimming/feature-switches.md) mode and that one relies on the SR type.

I think now I'm remembering why these were `[TEMPORARY EXCEPTION MESSAGE]` for such a long time...

Reply:
Yes, I'll pull these out into a separate file. Good point on the CoreRT impact. I had forgotten some of these details. Someone in the CoreRT repo will need to implement the resource lookup function in whatever way actually makes sense at runtime.